salt/elife-xpub/scripts/setup-database.sh

```diff
@@ -1,19 +1,21 @@
 #!/bin/bash
 set -e
 
-DC_COMMAND="docker-compose -f docker-compose.yml -f docker-compose.formula.yml"
-DB_CREATED_COMMAND="psql -c \"SELECT 'public.entities'::regclass\""
-DB_ENV="-e PGHOST=${PGHOST} -e PGPORT=${PGPORT} -e PGUSER=${PGUSER} -e PGDATABASE=${PGDATABASE} -e PGPASSWORD=${PGPASSWORD}"
-SETUP_ARGS="--username={{ pillar.elife_xpub.database.user }} --password={{ pillar.elife_xpub.database.password }} --email={{ pillar.elife_xpub.database.email }}"
+dc="docker-compose -f docker-compose.yml -f docker-compose.formula.yml"
+clean_migrate_command="scripts/clean-migrate-database.js"
+db_env="-e PGHOST=${PGHOST} -e PGPORT=${PGPORT} -e PGUSER=${PGUSER} -e PGDATABASE=${PGDATABASE} -e PGPASSWORD=${PGPASSWORD}"
+recreate="dropdb ${PGDATABASE} && createdb ${PGDATABASE}"
 
-if [ ! -z "${DROP}" ]; then
-    ${DC_COMMAND} run --rm app /bin/bash -c "npx pubsweet setupdb ${SETUP_ARGS} --clobber"
-    exit
-fi
+# Removing setting up a user (via clobber) in PubSweet as at the moment this is no longer necessary
+# This may be re-instated after the work to upgrade to the latest PubSweet version.
+# Ref: elifesciences/elife-xpub/issues/1920
+#
+# SETUP_ARGS="--username={{ pillar.elife_xpub.database.user }} --password={{ pillar.elife_xpub.database.password }} --email={{ pillar.elife_xpub.database.email }}"
 
-if ! ${DC_COMMAND} run --rm ${DB_ENV} postgres /bin/bash -c "${DB_CREATED_COMMAND}"
-then
-    ${DC_COMMAND} run --rm app /bin/bash -c "npx pubsweet setupdb ${SETUP_ARGS}"
+if [ ! -z "${DROP}" ]; then
+  ${dc} run --rm app /bin/bash -c "${clean_migrate_command}"
```

Comment:
this part of the script is not covered by the current tests of the formula, because the formula never passes in `DROP` (just saying)

Reply:
where are the tests for the formula? Is it somewhere near here:?
```
elife-alfred-formula/jenkinsfiles/Jenkinsfile.clean-end2end
71:                builderCmdNode 'elife-xpub--end2end', 1, 'cd /srv/elife-xpub && TIMEOUT=30 wait-database.sh && DROP=1 setup-database.sh'
```

Reply:
That is https://github.com/elifesciences/elife-alfred-formula/blob/19da7fc8a8d4217e06a69a5af5786d19b44fd263/jenkinsfiles/Jenkinsfile.clean-end2end#L71 but it isn't independently tested apart from running `process-clean-end2end`. Let me think if we could add a build variant to this repository that does that..

Reply:
Testing an experiment locally now, will push to this branch.